src/github.rs

```diff
@@ -9,6 +9,74 @@ use lazy_static::lazy_static;
 use regex::Regex;
 use serde::de::DeserializeOwned;
 
+/// A GitHub API client that optionally authenticates requests.
+pub struct Client {
+    github_api_token: Option<String>,
+}
+
+impl Client {
+    pub fn new(github_api_token: Option<String>) -> Self {
+        Self { github_api_token }
+    }
+
+    pub async fn public_organisation_members(
+        &self,
+        organisation: &str,
+    ) -> eyre::Result<HashSet<String>> {
+        #[derive(Debug, serde::Deserialize)]
+        pub struct Member {
+            login: String,
+        }
+
+        let url = format!(
+            "https://api.github.com/orgs/{}/members?per_page=100",
+            organisation
+        );
+        Ok(self
+            .api_list(url)
+            .await
+            .wrap_err("Unable to get public members for organisation")?
+            .into_iter()
+            .map(|member: Member| member.login)
+            .collect())
+    }
+
+    /// Perform a GET request to a paginated GitHub URL that returns a JSON array per
+    /// page. All pages will be traversed and retuned as a single collection.
+    async fn api_list<Json: DeserializeOwned>(&self, url: String) -> eyre::Result<Vec<Json>> {
+        let mut collection = Vec::new();
+        let mut next_url = Some(url);
+        while let Some(url) = next_url {
+            let response = self.api_get_response(&url).await?;
+            next_url = next_pagination_page(&response)?;
+            let items: Vec<Json> = response
+                .json()
+                .await
+                .wrap_err("Unable to parse JSON response")?;
+            collection.extend(items);
+        }
+
+        Ok(collection)
+    }
+
+    async fn api_get_response(&self, url: &str) -> eyre::Result<reqwest::Response> {
+        let request = reqwest::Client::new()
+            .get(url)
+            .header("accept", "application/vnd.github.v3+json")
+            .header("user-agent", "embark-oss");
+        let request = match &self.github_api_token {
+            Some(token) => request.header("authorization", format!("token {}", token)),
```

Comment:
here, `format!("token {}", token)),` shouldn't be used with `&`, we need to pass value by reference!